build/tar.go

```diff
@@ -0,0 +1,80 @@
+package build
+
+import (
+	"archive/tar"
+	"io"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+func tarDir(destinationFilename, sourceDir string) {
+
+	if destinationFilename[len(destinationFilename)-3:] != "tar" {
+		log.Fatalln("Please provide a valid tar filename")
+	}
+
+	tarFile, err := os.Create(destinationFilename)
+	checkError(err)
+
+	defer tarFile.Close()
+
+	var fileWriter io.WriteCloser = tarFile
+	tarfileWriter := tar.NewWriter(fileWriter)
+	defer tarfileWriter.Close()
+
+	walkTar(sourceDir, tarfileWriter)
+}
+
+func walkTar(dirPath string, tarfileWriter *tar.Writer) {
+
+	dir, err := os.Open(dirPath)
+	checkError(err)
+
+	dirInfo, err := dir.Stat()
+	checkError(err)
+
+	// prepare the tar header for dir entry.
+	dheader, err := tar.FileInfoHeader(dirInfo, "")
+	checkError(err)
+
+	dheader.Name = dir.Name()
+
+	err = tarfileWriter.WriteHeader(dheader)
+	checkError(err)
```

Comment:
Same. Remove `checkError(err)` and return the `error` and let the caller handle it.

Reply:
done